source/javascripts/controllers/_ProvProfileController.js.erb

```diff
@@ -163,28 +192,149 @@ angular.module("BitriseWorkflowEditor").controller("ProvProfileController", func
+		});
+	}
+
+	viewModel.capabilityDescriptionsOfProvProfile = function() {
+		return _.compact(_.uniq(_.map(viewModel.selectedProvProfile.capabilityKeys, function(aCapabilityKey) {
+			return ProvProfile.descriptionForCapabilityKey(aCapabilityKey);
+		})));
+	};
+
+	function validateTestDevices(provProfile) {
+		viewModel.validateTestDevicesProgress.start("<%= data[:strings][:code_signing][:prov_profile][:validate_test_devices][:in_progress] %>");
+
+		return requestService.fetchAllTestDevicesOfRepository().then(function(registeredTestDevices) {
+			_.each(provProfile.provisionedDevices, function(aProvisionedDevice) {
+				var registeredTestDeviceOnBitrise = _.find(registeredTestDevices.devices, function(aDevice) {
+					return aDevice.deviceIdentifier == aProvisionedDevice.udid;
+				});
+
+				if (registeredTestDeviceOnBitrise) {
+					aProvisionedDevice.statusCSSclass = "success";
+					aProvisionedDevice.registeredBy = registeredTestDeviceOnBitrise.owner.username;
+				}
+				else {
+					aProvisionedDevice.statusCSSclass = "error";
+					aProvisionedDevice.registeredBy = undefined;
+				}
+			});
+
+			var areAllValidationsSuccessfull = !_.any(provProfile.provisionedDevices, {
+				statusCSSclass: "error"
+			});
+
+			viewModel.provProfileConfigs[provProfile.databaseID].validationStatuses.deviceValidationStatus = areAllValidationsSuccessfull ? "success" : "error";
+
+			viewModel.validateTestDevicesProgress.success();
+		}, function(error) {
+			viewModel.validateTestDevicesProgress.error(error);
+		});
+	}
+
+	function validateCertificates(provProfile) {
+		viewModel.validateCertificatesProgress.start("<%= data[:strings][:code_signing][:prov_profile][:validate_certificates][:in_progress] %>");
+
+		if (certificateService.progress.isInProgress) {
+			$timeout(function() {
+				validateCertificates(provProfile);
+			}, validateCertificatesReloadIntervalInSeconds * 1000);
+		}
+		else {
+			_.each(provProfile.developerCertificateIdentities, function(anIdentity) {
+				if (certificateService.progress.isError) {
+					anIdentity.statusCSSclass = "error";
+
+					return false;
+				}
+				else {
+					var isIdentityRegistered = _.any(certificateService.certificateDetailStateMaps, function(aCertificateDetailStateMapsItem) {
+						return _.find(aCertificateDetailStateMapsItem.certificate.details, function(aCertificateIdentity) {
+							return aCertificateIdentity.serial == anIdentity.serial;
+						});
+					});
+
+					anIdentity.statusCSSclass = isIdentityRegistered ? "success" : "error";
+						
+					return isIdentityRegistered;
+				}
+			});
+
+			var areAllValidationsSuccessfull = !_.any(provProfile.developerCertificateIdentities, {
+				statusCSSclass: "error"
+			});
+
+			viewModel.provProfileConfigs[provProfile.databaseID].validationStatuses.certificateValidationsStatus = areAllValidationsSuccessfull ? "success" : "error";
+
+			viewModel.validateCertificatesProgress.success();
+		}
+
+		return $q.when();
```

Comment:
This will always return a fulfilled promise, a bit misleading, especially since "validation" should not be considered as successful when e. g. isIdentityRegistered is false, or areAllValidationsSuccessfull is false.

If we don't care about this validation's success/fail, we should structure the code differently: don't include it in the $q.all of `validateDetails`, simply call it there.